cmd/poller/plugin/label_agent/label_agent.go

```diff
@@ -192,12 +201,63 @@ func (me *LabelAgent) excludeRegex(instance *matrix.Instance) {
 	for _, r := range me.excludeRegexRules {
 		if r.reg.MatchString(instance.GetLabel(r.label)) {
 			instance.SetExportable(false)
-			me.Logger.Trace().Msgf("excludeEquals: (%s) [%s] instance with labels [%s] => excluded", r.label, r.reg.String(), instance.GetLabels().String())
+			me.Logger.Trace().Str("label", r.label).
+				Str("regex", r.reg.String()).
+				Str("instance labels", instance.GetLabels().String()).
+				Msg("excludeRegex: excluded")
 			break
 		}
 	}
 }
 
+// if label is not equal to value, set instance as non-exportable
+func (me *LabelAgent) includeEquals(instance *matrix.Instance) {
+	isExport := false
+	for _, r := range me.includeEqualsRules {
+		if instance.GetLabel(r.label) == r.value {
+			isExport = true
+			me.Logger.Trace().Str("label", r.label).
+				Str("value", r.value).
+				Str("instance labels", instance.GetLabels().String()).
+				Msg("includeEquals: included")
+			break
+		}
+	}
+	instance.SetExportable(isExport)
+}
+
+// if label does not contains value, set instance as non-exportable
+func (me *LabelAgent) includeContains(instance *matrix.Instance) {
+	isExport := false
+	for _, r := range me.includeContainsRules {
+		if strings.Contains(instance.GetLabel(r.label), r.value) {
+			isExport = true
+			me.Logger.Trace().Str("label", r.label).
+				Str("value", r.value).
+				Str("instance labels", instance.GetLabels().String()).
+				Msg("includeContains: included")
+			break
+		}
+	}
+	instance.SetExportable(isExport)
+}
+
+// if label does not equal to value, set instance as non-exportable
```

Comment:
// if label does not match regex, do not export the instance
or with fewer negatives
// only export instances with a matching (regex) label
// if an instance does not match the regex label it will not be exported